xmlhttp.php

```diff
@@ -281,6 +281,9 @@
 		}
 	}
 
+	// Filter out duplicates, if any, from recursive array data
+	$data = array_intersect_key($data, array_unique(array_map('serialize', $data)));
```

Comment:
This should use `my_serialize` rather than plain old `serialize`: https://github.com/mybb/mybb/blob/feature/inc/functions.php#L2289

That way things like multibyte strings and such are taken care of, which could be possible for user names.

Reply:
Done.